Client/Frontend/Browser/TabTrayController.swift

```diff
@@ -644,7 +644,17 @@ extension TabTrayController: TabManagerDelegate {
         // check this before removing the item from the collection
         let removedIndex = tabDataSource.removeTab(tab)
         if removedIndex > -1 {
-            self.collectionView.deleteItemsAtIndexPaths([NSIndexPath(forItem: removedIndex, inSection: 0)])
+            self.collectionView.performBatchUpdates({
+                self.collectionView.deleteItemsAtIndexPaths([NSIndexPath(forItem: removedIndex, inSection: 0)])
+            }, completion: { finished in
+                if finished {
```

Comment:
nit: in a very personal sense of hating very nested if statements, how about something like:
```
if #available(iOS 9, *) {
        guard finished && self.privateTabsAreEmpty() else { return }
        self.emptyPrivateTabsView.hidden = false
}
```?